modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java

```diff
@@ -295,13 +313,23 @@ private void addCatchAllPathMapIfNotNull(PathMap currentPathMap, Stack<PathMap>
             }
         }
 
-        private RequestHandlerMatcherPair findRequestHandlerMatcherPair(List<RequestHandlerMatcherPair> requestHandlerMatcherPairs, HttpRequest request)
+        private RequestHandlerMatcherPair findRequestHandlerMatcherPair(List<RequestHandlerMatcherPair> requestHandlerMatcherPairs, HttpRequest request, String path)
         {
+            final String[] pathParts = splitPath(path);
             for (RequestHandlerMatcherPair requestHandlerMatcherPair : requestHandlerMatcherPairs)
             {
                 if (requestHandlerMatcherPair.getRequestMatcher().matches(request))
                 {
-                    return requestHandlerMatcherPair;
+                    String requestHandlerPath = requestHandlerMatcherPair.getRequestMatcher().getPath();
+                    if(requestHandlerPath.contains("*") && ! requestHandlerPath.endsWith("*") && ! requestHandlerPath.endsWith("*/") ){
+                        String vector [] = requestHandlerMatcherPair.getRequestMatcher().getPath().split("\\*");
+                        for(String pathPart : pathParts){
+                            if(vector.length>0 && ! pathPart.equals("") && vector[1].contains(pathPart))
```

Comment:
No. They are differents variables.
Vector contains the differents parts of the path of the requestMatcher.
pathPaths contains the differents parts of the current request path.